source/code/plugins/patch_management_lib.rb

```diff
@@ -11,9 +12,13 @@ class LinuxUpdates
     @@prev_hash = ""
     @@delimiter = "_"
     @@force_send_last_upload = Time.now
+
```

Comment:
@@log is not initialized, and it's better to be an instance member (@log), and initialized in the constructor. 
prev_hash, and force_send_last_upload blocks the filter being used multiple times, since all the filter instances will share the same static member. it's better to have a persistent file in the filter conf, and the user can configure it.